packages/ripple/src/index.ts

```diff
@@ -0,0 +1,127 @@
+import { getRippleGlobalNamespace } from './getRippleGlobalNamespace';
+import { Options, colorMap, Mode } from './utils';
+
+const TRANSITION_TIME = 300;
+const RIPPLE_NAMESPACE = getRippleGlobalNamespace();
+
+function rippleEvent(event: MouseEvent) {
+  if (
+    RIPPLE_NAMESPACE?.registeredRippleElements.has(event.target as HTMLElement)
+  ) {
+    createRippleEffect(event);
+  }
+}
+
+export function registerRipple(node: HTMLElement, options: Options) {
+  if (!RIPPLE_NAMESPACE) {
+    return;
+  }
+
+  RIPPLE_NAMESPACE.registeredRippleElements.set(node, options);
+
+  if (!RIPPLE_NAMESPACE.setRippleListener) {
+    document.addEventListener('click', rippleEvent, { passive: true });
+
+    const styles = document.createElement('style');
+    styles.innerHTML = staticRippleStyles;
+    document.head.append(styles);
+
+    RIPPLE_NAMESPACE.setRippleListener = true;
+  }
+
+  return unregisterRipple;
```

Comment:
We can make this an anonymous function that captures the reference to the node such that an application only has to call the function - they don't have to pass in the node itself. Makes for tight coupling of the unregister method.

Reply:
👍  makes sense to me